### DISCUSSION TITLE
What is the proper way to generate large (multiple chunks) structures?

### DISCUSSION THREAD

**Discussion by ghost** [Minestom/Minestom#1859] [Help]
My initial idea was to check if given GenerationUnit is located within the area where the structure is supposed to be and whether the structure at this "slot" is already marked as generated, and if conditions are met - fork it, such that it covers the whole area of the structure (multiple chunks), generate the structure and mark it as generated. The problem is that it seems like there's no way to get the absolute coordinates of the GenerationUnit.

What I'm trying to do is generate a fixed number of islands of a fixed size at a fixed distance.

```
ISLAND.  ISLAND   ISLAND

ISLAND   ISLAND   ISLAND

ISLAND   ISLAND   ISLAND
```

Is it a bad idea to use the Minestom generator for this? Should I use some other tool to "pre-generate" the map?

**Comment by ghost** [comment]
Yeah, I just realized how dumb it is to use the generator to create a fixed number of structures at the fixed coordinates. I should just create them manually whenever I want to initialize the instance.

**Reply by KrystilizeNevaDies** [reply]
I don't think it's a bad idea depending on what you are looking for :)

But you should look at the unit#fork(Consumer) method for implementing this.

**Reply by ghost** [reply]
> I don't think it's a bad idea depending on what you are looking for :)
> 
> But you should look at the unit#fork(Consumer) method for implementing this.

My current understanding is that the generator should be used for procedural world generation only, not for unique stuff. Now I'm just building structures on a separate server with WorldEdit and transferring them to main via schematics (thanks to this library https://github.com/hollow-cube/common/tree/main/modules/schem which should definitely get more attention). 

I have a quick question though (don't wanna create another discussion): Is there a way to disable automatic chunk loading or limit the view distance for players?

**Reply by KrystilizeNevaDies** [reply]
Yes there is a InstanceContainer#setAutoViewable
